src/libz.jl

```diff
@@ -23,6 +23,18 @@ mutable struct ZStream
     reserved::Culong
 end
 
+@assert typemax(Csize_t) ≥ typemax(Cuint)
+
+function zalloc(::Ptr{Cvoid}, items::Cuint, size::Cuint)::Ptr{Cvoid}
+    s, f = Base.Checked.mul_with_overflow(items, size)
+    if f
+        C_NULL
+    else
+        ccall(:jl_malloc, Ptr{Cvoid}, (Csize_t,), s%Csize_t)
```

Comment:
Is there a reason we use `ccall` here rather than `Libc.malloc` ?

Reply:
This is to make sure the memory being used is visible to the GC https://discourse.julialang.org/t/c-struct-garbage-collection-not-run-frequently-enough/116919/14

If I replace this with `Libc.malloc` Julia uses more memory in the memory leak test, because GC isn't run frequently enough.

Reply:
There is some reference within the C code suggesting that `calloc` may be warranted here.

https://github.com/madler/zlib/blob/develop/deflate.c#L397

The comments here suggest that malloc may be an appropriate optimization on modern systems.

https://github.com/madler/zlib/issues/517